Compilation error in XMLHandler.h

Dear OpenMS team,

we are getting this error when compiling our software with latest develop branch of OpenMS.

```
/root/OpenMS/src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h: In static member function 'static OpenMS::Internal::unique_xerces_ptr<char16_t> OpenMS::Internal::StringManager::fromNative_(const char*)':
/root/OpenMS/src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h:230:33: error: missing template arguments before '(' token
  230 |         return unique_xerces_ptr(xercesc::XMLString::transcode(str));
```

Indeed the template argument is missing there.

Also here, we get the same error here:
```
      inline static String toNative_(const XMLCh* str)
      {
        return unique_xerces_ptr(xercesc::XMLString::transcode(str)).get();
      }
```

the implementation of this one should be replaced with (?)
```
      inline static String toNative_(const XMLCh* str)
      {
        return String(xercesc::XMLString::transcode(str));
      }
```

doing so, another issue appears: compiler complains about using deleted methods (copy constructors and assignment) which can be fixed adding move constructor for unique_ptr.

```
      unique_xerces_ptr(const unique_xerces_ptr&& other) noexcept 
        : item_(nullptr)
      {
        this->swap(other);
      }; 
      void swap(unique_xerces_ptr<T>& other) noexcept
      {
        using std::swap;
        swap(item_, other.item_);
      }
```

these samples of code are not fully tested, it just allows us to compile with OpenMS.

we got this error on both ubuntu:20.04
The CXX compiler identification is GNU 9.3.0

and MacOS
compiler identification is AppleClang 11.0.0.11000033

note that OpenMS compile fine in these platforms.
problems occur when we include the file:

```
In file included from /root/OpenMS/src/openms/include/OpenMS/FORMAT/QcMLFile.h:38,
                 from /root/OpenMS/src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h:40,
                 from /root/SmartPeak/src/smartpeak/include/SmartPeak/core/RawDataProcessor.h:29,
                 from /root/SmartPeak/src/smartpeak/include/SmartPeak/core/ApplicationHandler.h:27,
                 from /root/SmartPeak/src/smartpeak/include/SmartPeak/core/ApplicationProcessor.h:26,
                 from /root/SmartPeak/src/smartpeak/source/core/ApplicationProcessor.cpp:24:
/root/OpenMS/src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h: In static member function 'static OpenMS::Internal::unique_xerces_ptr<char16_t> OpenMS::Internal::StringManager::fromNative_(const char*)':
/root/OpenMS/src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h:230:33: error: missing template arguments before '(' token
  230 |         return unique_xerces_ptr(xercesc::XMLString::transcode(str));
```

--- a/src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h
+++ b/src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h
@@ -147,8 +147,8 @@ namespace OpenMS
     public:
 
       // Hide copy constructor and assignment operator
-      unique_xerces_ptr(const unique_xerces_ptr&) = delete;
-      unique_xerces_ptr& operator=(const unique_xerces_ptr&) = delete;
+      unique_xerces_ptr(const unique_xerces_ptr<T>&) = delete;
+      unique_xerces_ptr& operator=(const unique_xerces_ptr<T>&) = delete;
 
       unique_xerces_ptr()
           : item_(nullptr)
@@ -161,6 +161,17 @@ namespace OpenMS
       ~unique_xerces_ptr()
       {
         xerces_release();
+      }
+
+      unique_xerces_ptr(const unique_xerces_ptr<T>&& other) noexcept
+          : item_(nullptr)
+      {
+        this->swap(other);
+      }
+
+      void swap(unique_xerces_ptr<T>& other) noexcept
+      {
+        std::swap(item_, other.item_);
       }
 
       // Assignment of data to guard (not chainable)
@@ -227,7 +238,7 @@ namespace OpenMS
       // Converts from a narrow-character string to a wide-character string.
       inline static unique_xerces_ptr<XMLCh> fromNative_(const char* str)
       {
-        return unique_xerces_ptr(xercesc::XMLString::transcode(str));
+        return unique_xerces_ptr<XMLCh>(xercesc::XMLString::transcode(str));
       }
 
       // Converts from a narrow-character string to a wide-character string.
@@ -239,7 +250,7 @@ namespace OpenMS
       // Converts from a wide-character string to a narrow-character string.
       inline static String toNative_(const XMLCh* str)
       {
-        return unique_xerces_ptr(xercesc::XMLString::transcode(str)).get();
+        return String(unique_xerces_ptr<char>(xercesc::XMLString::transcode(str)).get());
       }
 
       // Converts from a wide-character string to a narrow-character string.

--- a/src/openms/source/FORMAT/HANDLERS/XQuestResultXMLHandler.cpp
+++ b/src/openms/source/FORMAT/HANDLERS/XQuestResultXMLHandler.cpp
@@ -217,7 +217,7 @@ namespace OpenMS
 
     void XQuestResultXMLHandler::endElement(const XMLCh * const /*uri*/, const XMLCh * const /*local_name*/, const XMLCh * const qname)
     {
-      String tag = XMLString::transcode(qname);
+      String tag = StringManager::convert(qname);
       if (tag == "xquest_results")
       {
         if (!this->is_openpepxl_)
@@ -237,7 +237,7 @@ namespace OpenMS
 
     void XQuestResultXMLHandler::startElement(const XMLCh * const, const XMLCh * const, const XMLCh * const qname, const Attributes &attributes)
     {
-      String tag = XMLString::transcode(qname);
+      String tag = StringManager::convert(qname);
       // Extract meta information from the xquest_results tag
       if (tag == "xquest_results")
       {

